cpu/lm4f120/include/periph_cpu.h

```diff
@@ -106,6 +106,27 @@ typedef enum {
 } adc_res_t;
 #endif /* ndef DOXYGEN */
 
+/**
+ * @name   UART device configuration
+ * @{
+ */
+typedef struct {
+    unsigned long dev;                 /**< UART device number */
+    unsigned long uart_txint_mode;     /**< UART TX interrupt mode */
+    unsigned long uart_fifo_tx;        /**< UART TX fifo select */
+    unsigned long uart_fifo_rx;        /**< UART RX fifo select */
+    IRQn_Type uart_irq_chan;           /**< UART IRQ channel */
+    unsigned long gpio_sysctl;         /**< GPIO device in sysctl */
+    unsigned long gpio_port;           /**< GPIO port */
+    struct {
+        unsigned long rx;              /**< pin used for RX */
+        unsigned long tx;              /**< pin used for TX */
+        unsigned long mask_rx;         /**< Pin mask for RX*/
+        unsigned long mask_tx;         /**< Pin mask for TX*/
+    } pins;                            /**< Pin setting */
+} uart_conf_t;
```

Comment:
The whole structure has >10 "unsigned longs". Do all of the fields need to be 32bit wide?

Reply:
Everything outside of the .dev and uart_irq_chan is vendor defined as 32 bits.  All things that are in the uart_config_t are either user configurable or requires ROM mapping.  I am talking with @smlng regarding hiding mapping in the .c file or leaving it in the board file.  It is a tradeoff question for configurablility vs user ease.

Reply:
ok!